### PR TITLE
feat: optmised gear staking

### PR DESCRIPTION
**Pull request by 0xmikko**

### CODE REVIEW

File: contracts/interfaces/ICreditManagerV3.sol [hunk 1]
```diff
@@ -46,7 +46,6 @@ struct CollateralDebtData {
     uint256 cumulativeIndexNow;
     uint256 cumulativeIndexLastUpdate;
     uint128 cumulativeQuotaInterest;
```
Comment (lekhovitsky) [inline contracts/interfaces/ICreditManagerV3.sol:48]: `uint128` -> `uint256` for uniformity

File: contracts/credit/CreditManagerV3.sol [hunk 2]
```diff
@@ -817,7 +819,7 @@ contract CreditManagerV3 is ICreditManagerV3, SanityCheckTrait, ReentrancyGuardT
         }) + collateralDebtData.cumulativeQuotaInterest; // U:[CM-21] // I: [CMQ-07]
 
         collateralDebtData.accruedFees = (collateralDebtData.accruedInterest * feeInterest) / PERCENTAGE_FACTOR
-            + (supportsQuotas ? collateralDebtData.quotaProfits : 0); // U:[CM-21]
+            + (supportsQuotas ? creditAccountInfo[creditAccount].quotaProfits : 0); // U:[CM-21]
```
Comment (lekhovitsky) [inline contracts/credit/CreditManagerV3.sol:822]: should it still be `quotaProfits`?

File: contracts/credit/CreditManagerV3.sol [hunk 3]
```diff
@@ -471,19 +471,22 @@ contract CreditManagerV3 is ICreditManagerV3, SanityCheckTrait, ReentrancyGuardT
 
             uint128 newCumulativeQuotaInterest;
             uint128 newQuotaProfits;
+
```
Comment (lekhovitsky) [inline contracts/credit/CreditManagerV3.sol:474]: check extra empty lines

File: contracts/support/GearStakingV3.sol [hunk 4]
```diff
@@ -260,8 +184,47 @@ contract GearStakingV3 is ACLNonReentrantTrait, IGearStakingV3 {
...
+
+    /// @notice Returns the current global voting epoch
+    function getCurrentEpoch() public view returns (uint16) {
+        if (block.timestamp < firstEpochTimestamp) return 0; // U:[GS-01]
+        return uint16((block.timestamp - firstEpochTimestamp) / EPOCH_LENGTH) + 1; // U:[GS-01]
+    }
+
+    /// @notice Returns the total amount of GEAR the user staked into staked GEAR
+    function balanceOf(address user) external view returns (uint256) {
+        return uint256(voteLockData[user].totalStaked);
```
Comment (lekhovitsky) [inline contracts/support/GearStakingV3.sol]: no need to cast

**Review by lekhovitsky — APPROVED** [review]

File: contracts/interfaces/ICreditManagerV3.sol (hunk 1 above)
Comment (0xmikko) [reply contracts/interfaces/ICreditManagerV3.sol:48]: it'll be removed soon for gas savings

**Pull request merged**